### PR TITLE
Upgrade cryptography to 3.4.6

### PR DESCRIPTION
**Pull request by ohemorange**
Docker builds (including arm) are passing, along with the rest of the extended tests:
https://dev.azure.com/certbot/certbot/_build/results?buildId=3609&view=results

Snap builds are passing (yes, definitely with these changes):
https://dev.azure.com/certbot/certbot/_build/results?buildId=3625&view=results

Test farm tests are passing:
https://dev.azure.com/certbot/certbot/_build/results?buildId=3639&view=results

Between those three runs (streamlined for efficiency), that's all the extended tests.

### CODE REVIEW

**Comment by bmw** [comment]
Just taking a quick look at the changes in this PR, it looks great to me, but it looks like the `test_sdist.sh` test farm test timed out after an hour.

**Comment by ohemorange** [comment]
Yeah, working on it, might be a little annoying to fix the most cleanly or a little time-consuming on runs to fix easily.

**Comment by bmw** [comment]
Thanks. I hesitated to point out the test failure because I figured you were watching it, but I wanted to write a quick comment to make sure we're all on the same page with this PR.

I hope fixing it isn't too bad. If it helps, I personally think we can consider things like dropping tests on old AMIs.

**Comment by ohemorange** [comment]
>  If it helps, I personally think we can consider things like dropping tests on old AMIs.

Unfortunately not really; unless "debian stable" counts as an old AMI for you :D

**Comment by bmw** [comment]
Ah yeah that's a stretch. Assuming it's the ARM one, maybe we can argue that making that work is a poor use of our time, but I'd personally want to know what the problem is with the test and why we added the ARM build to `test_sdists.sh` in the first place.

**Comment by ohemorange** [comment]
Nope, it's both. Debian stable just doesn't have a new enough rust packaged at all.

**Comment by ohemorange** [comment]
IIRC, we added that arm build so we're testing on at least one arm build; I think we could switch it to a different distro, but again, that's not the issue.

**Comment by ohemorange** [comment]
https://cryptography.io/en/latest/installation.html#debian-ubuntu

File: tools/snap/generate_dnsplugins_snapcraft.sh [hunk 1]
```diff
@@ -35,7 +35,15 @@ parts:
       - PIP_CONSTRAINT: \$SNAPCRAFT_PART_SRC/snap-constraints.txt
       - SNAP_BUILD: "True"
     # To build cryptography and cffi if needed
-    build-packages: [gcc, libffi-dev, libssl-dev, python3-dev]
+    build-packages:
+      - gcc
+      - git
+      - libaugeas-dev
```
Comment (bmw) [inline tools/snap/generate_dnsplugins_snapcraft.sh:41]: nit: I don't think we need Augeas in our DNS plugins.

**Review by bmw — CHANGES_REQUESTED** [review]
I noticed one small thing but this otherwise LGTM. Thanks for working on this.

**Review by bmw — APPROVED** [review]

File: tools/snap/generate_dnsplugins_snapcraft.sh (hunk 1 above)
Comment (ohemorange) [reply tools/snap/generate_dnsplugins_snapcraft.sh]: lollll yes I did just copy and paste from the certbot snapcraft file

Comment (bmw) [reply tools/snap/generate_dnsplugins_snapcraft.sh]: Makes sense. I almost didn't notice it either.

**Pull request merged**